spark-operator/src/main/java/org/apache/spark/k8s/operator/config/SparkOperatorConf.java

```diff
@@ -0,0 +1,429 @@
+          .defaultValue(120L)
+          .typeParameterClass(Long.class)
+          .build();
+
+  public static final ConfigOption<Boolean> TRIM_ATTEMPT_STATE_TRANSITION_HISTORY =
+      ConfigOption.<Boolean>builder()
+          .key("spark.kubernetes.operator.reconciler.trimStateTransitionHistoryEnabled")
+          .description(
+              "When enabled, operator would trim state transition history when a "
+                  + "new attempt starts, keeping previous attempt summary only.")
+          .defaultValue(true)
+          .typeParameterClass(Boolean.class)
+          .build();
+
+  public static final ConfigOption<String> SPARK_APP_STATUS_LISTENER_CLASS_NAMES =
+      ConfigOption.<String>builder()
+          .key("spark.kubernetes.operator.reconciler.appStatusListenerClassNames")
+          .defaultValue("")
+          .description("Comma-separated names of SparkAppStatusListener class implementations")
+          .enableDynamicOverride(false)
+          .typeParameterClass(String.class)
+          .build();
+
+  public static final ConfigOption<Boolean> DYNAMIC_CONFIG_ENABLED =
+      ConfigOption.<Boolean>builder()
+          .key("spark.kubernetes.operator.dynamicConfig.enabled")
+          .typeParameterClass(Boolean.class)
+          .description(
+              "When enabled, operator would use config map as source of truth for config "
+                  + "property override. The config map need to be created in "
+                  + "spark.kubernetes.operator.namespace, and labeled with operator name.")
+          .defaultValue(false)
+          .enableDynamicOverride(false)
+          .build();
+
+  public static final ConfigOption<String> DYNAMIC_CONFIG_SELECTOR_STR =
+      ConfigOption.<String>builder()
+          .key("spark.kubernetes.operator.dynamicConfig.selectorStr")
+          .typeParameterClass(String.class)
+          .description("The selector str applied to dynamic config map.")
+          .defaultValue(
+              SparkReconcilerUtils.labelsAsStr(SparkReconcilerUtils.defaultOperatorConfigLabels()))
+          .enableDynamicOverride(false)
+          .build();
+
+  public static final ConfigOption<Integer> DYNAMIC_CONFIG_RECONCILER_PARALLELISM =
+      ConfigOption.<Integer>builder()
+          .key("spark.kubernetes.operator.dynamicConfig.reconcilerParallelism")
+          .description("Parallelism for dynamic config reconciler. Use -1 for unbounded pool.")
+          .enableDynamicOverride(false)
+          .typeParameterClass(Integer.class)
+          .defaultValue(1)
+          .build();
+
+  public static final ConfigOption<Integer> RATE_LIMITER_REFRESH_PERIOD_SECONDS =
+      ConfigOption.<Integer>builder()
+          .key("spark.kubernetes.operator.rateLimiter.refreshPeriodSeconds")
+          .description("Operator rate limiter refresh period(in seconds) for each resource.")
+          .enableDynamicOverride(false)
+          .typeParameterClass(Integer.class)
+          .defaultValue(15)
```

Comment:
Since this is `Integer.class`, what happens for the corner values like `0` or `-1`?